### ISSUE TITLE
LLVM name mangling bypass \x01 seems broken

### ISSUE THREAD

**Issue by MauriceKayser** [rust-lang/rust#110294]
I think https://github.com/rust-lang/rust/pull/96444 might have broken the [#[export_name = "\x01..."]](https://github.com/rust-lang/rust/issues?q=%5Cx01+mangle) LLVM workaround for avoiding the "@XYZ" name mangling on x86. Compiling the following with `i686-pc-windows-msvc` results in `error LNK2001: unresolved external symbol _☺test1@0`:

```rust
#[export_name = "\x02test2"]
extern "stdcall" fn test2() {}

#[export_name = "\x01test1"]
extern "stdcall" fn test1() {}
```
I stumbled upon this when trying to use the `\x01` workaround to specify the `_aulldiv` symbol, which the MSVC linker expects to be named `__aulldiv` (instead of `__aulldiv@16`) in `#![no_std]` x86 Windows crates.

**Comment by nbdd0121** [comment]
Yeah, my PR indeed doesn't special case `\x01`. I believe the `\x01` thing is a special LLVM-specific hack and was never the proper solution. We probably want a proper solution to turn off symbol decoration in Windows.

**Comment by nbdd0121** [comment]
Are you trying to define or reference that symbol? You can use asm to workaround the issue.

**Comment by MauriceKayser** [comment]
> We probably want a proper solution to turn off symbol decoration in Windows.

Agreed, there are other issues with the `export_name` / `no_mangle` architecture on Windows as well, like https://github.com/rust-lang/rust/issues/67399.

> Are you trying to define or reference that symbol? You can use asm to workaround the issue.

I am trying to get rid of the [naked_functions](https://github.com/rust-lang/rust/issues/32408) feature in my crates ([1](https://github.com/MauriceKayser/rs-windows-builtins/blob/master/no_std/builtins/src/x86.rs), [2](https://github.com/MauriceKayser/rs-winapi2/blob/master/src/builtins.rs)), by porting the naked assembly functions to Rust code. The naked function does not care about the calling convention, but the Rust implementation does.

**Comment by nbdd0121** [comment]
If you are getting rid of naked functions to remove nightly dependency, then you can switch to `global_asm`. If you are trying to reimplement the function in Rust, then what you could do is to add a `global_asm!` stub that jumps to the Rust implementation.

**Comment by apiraino** [comment]
WG-prioritization assigning priority ([Zulip discussion](https://rust-lang.zulipchat.com/#narrow/stream/245100-t-compiler.2Fwg-prioritization.2Falerts/topic/.23110294.20LLVM.20name.20mangling.20bypass.20.5Cx01.20seems.20broken)).

@rustbot label -I-prioritize +P-low
